### PR TITLE
[TA] change the way OM is set and improve docstrings

### PR DESCRIPTION
**Pull request by maririos**
Based on the Ux study for Opinion mining, we want to make OM easier to discover so we are switching from Flag enum to bool in the `AnalyzeSentimentOptions` class.

It also updates docstrings to include more useful information, format to add paragraphs and hints.

### CODE REVIEW

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs [hunk 1]
```diff
@@ -453,18 +453,18 @@ public virtual async Task<Response<CategorizedEntityCollection>> RecognizeEntiti
...
-        /// such as person, location, or organization.  For more information on
-        /// available categories, see
-        /// <a href="https://docs.microsoft.com/en-us/azure/cognitive-services/Text-Analytics/named-entity-types"/>.
-        /// For a list of languages supported by this operation, see
-        /// <a href="https://docs.microsoft.com/en-us/azure/cognitive-services/text-analytics/language-support"/>.
-        /// For document length limits, maximum batch size, and supported text encoding, see
-        /// <a href="https://docs.microsoft.com/azure/cognitive-services/text-analytics/overview#data-limits"/>.
+        /// such as person, location, or organization.
+        /// <para>For more information on available categories, see
+        /// <a href="https://docs.microsoft.com/en-us/azure/cognitive-services/Text-Analytics/named-entity-types"/>.</para>
```
Comment (maririos) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs:458]: create issue https://github.com/Azure/azure-sdk-for-net/issues/16280 to do more cleanup

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeSentimentOptions.cs [hunk 2]
```diff
@@ -5,13 +5,14 @@ namespace Azure.AI.TextAnalytics
 {
     /// <summary>
     /// Options that allow callers to specify details about how the operation
-    /// is run and what information is returned from it by the service.
+    /// is run. For example set model version, whether to include statistics, and execute opinion mining.
```
Comment (iscai-msft) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeSentimentOptions.cs:8]: I would include opinion mining first in this list of examples to have it jump out more

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeSentimentResultCollection.cs [hunk 3]
```diff
@@ -20,8 +20,8 @@ internal AnalyzeSentimentResultCollection(IList<AnalyzeSentimentResult> list, Te
 
         /// <summary>
         /// Gets statistics about the documents and how it was processed
-        /// by the service.  This property will have a value when IncludeStatistics
-        /// is set to true in the client call.
+        /// by the service.  This property will have a value when IncludeStatistics in <see cref="AnalyzeSentimentOptions"/>
+        /// is set to True in the client call.
```
Comment (iscai-msft) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeSentimentResultCollection.cs:24]: should be lower case for .net

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/LinkedEntity.cs [hunk 4]
```diff
@@ -62,7 +62,7 @@ internal LinkedEntity(string name, IEnumerable<LinkedEntityMatch> matches, strin
 
         /// <summary> Bing Entity Search API unique identifier of the recognized entity.
         /// Use in conjunction with the Bing Entity Search API to fetch additional relevant information.
-        /// This property is only available for v3.1-preview.2 and up.
+        /// This property is only available for service v3.1-preview.2 and up.
```
Comment (iscai-msft) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/LinkedEntity.cs:65]: nit: service API version v3.1-preview.2 and up.

I've also noticed you refer to the latest preview version as both the generic v3.1-preview, and v3.1-preview.2. Don't have a strong preference over which one you use, but think they should be consistent. Python uses v3.1-preview to refer to latest service version

**Review by iscai-msft — APPROVED** [review]

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/LinkedEntity.cs (hunk 4 above)
Comment (maririos) [reply sdk/textanalytics/Azure.AI.TextAnalytics/src/LinkedEntity.cs:65]: Fair. modified to `v3.1-preview and up.`

**Pull request merged**